File: Level 0/주사위 게임 3/solution.py
def solution(a, b, c, d):
    nums = [a, b, c, d]
    counts = [nums.count(num) for num in nums]
    maxCount = max(counts)

    if maxCount == 4:
        return 1111 * a
    if maxCount == 3:
        return (10 * nums[counts.index(3)] + nums[counts.index(1)]) ** 2
    if maxCount == 2:
        if min(counts) == 2:
            p, q = max(nums), min(nums)
            return (p + q) * (p - q)
        else:
            p = nums[counts.index(2)]
            return (a * b * c * d) / p**2
    if maxCount == 1:
        return min(nums)

File: Level 0/주사위 게임 3/solution.js
function solution(a, b, c, d) {
  const map = new Map();

  [a, b, c, d].forEach((num) => {
    map.set(num, (map.get(num) || 0) + 1);
  });

  const arr = [...map.keys()].sort((a, b) => map.get(b) - map.get(a));
  const maxCount = Math.max(...map.values());
  const [p, q, r, s] = arr;

  switch (arr.length) {
    case 1:
      return 1111 * p;
    case 2:
      return maxCount === 2 ? (p + q) * Math.abs(p - q) : (10 * p + +q) ** 2;
    case 3:
      return q * r;
    case 4:
      return Math.min(p, q, r, s);
  }
}
